Update dependencies (NPM and others)

deribaucourt opened the pull request:
Run `npm update` in every package to update the dependencies. Fixes a few security audit warnings.

WilsonZiweiWang committed:
Auto update references for commits, tags, etc

WilsonZiweiWang committed:
Auto update references for commits, tags, etc

WilsonZiweiWang committed:
Auto update tree-sitter-bash parser info

WilsonZiweiWang committed:
Auto update tree-sitter-bash parser info

deribaucourt committed:
npm: update npm packages

Run `npm update` in every package to update the dependencies.
Fixes a few security audit warnings.

deribaucourt committed:
npm: update npm packages

Run `npm update` in every package to update the dependencies.
Fixes a few security audit warnings.

deribaucourt committed:
scripts: update references and versions

Run our update-refs scripts to bump all non-nodejs dependencies.

deribaucourt committed:
scripts: update references and versions

Run our update-refs scripts to bump all non-nodejs dependencies.

deribaucourt committed:
integration-tests: update for electron/vscode-test

Update the run-test script based on the latest vscode-test sample.

deribaucourt committed:
integration-tests: update for electron/vscode-test

Update the run-test script based on the latest vscode-test sample.

deribaucourt committed:
integration-tests: update for electron/vscode-test

Update the run-test script based on the latest vscode-test sample.

deribaucourt commented:
@WilsonZiweiWang from my experience, some tests failed due to calling `devtool modify`. With the updated poky, it seemed to download Uninative and build many recipes to patch sources which is extremly slow.
It looked like if we stay on the current poky version we were unaffected. It would be ok for now.

Maybe I'm wrong and the tests fail for a reason, but that's my thoughts so far.

WilsonZiweiWang commented:
> @WilsonZiweiWang from my experience, some tests failed due to calling `devtool modify`. With the updated poky, it seemed to download Uninative and build many recipes to patch sources which is extremly slow. It looked like if we stay on the current poky version we were unaffected. It would be ok for now.
> 
> Maybe I'm wrong and the tests fail for a reason, but that's my thoughts so far.

When the integration tests are skipped, the CI can consistently pass if a little more time is given to the unit tests, for example, 10 minutes of timeout. 

Successful runs with integration tests skipped:
https://github.com/yoctoproject/vscode-bitbake/actions/runs/17841666750/job/50732404783
https://github.com/yoctoproject/vscode-bitbake/actions/runs/17841666750/job/50734311096
https://github.com/yoctoproject/vscode-bitbake/actions/runs/17841666750/job/50734781920

According to the logs from a previous CI run and my local runs, they are all stuck at the `reviewDiagnostics()` from the integration tests. See this [run](https://github.com/yoctoproject/vscode-bitbake/actions/runs/17838149329/job/50720468929) where the integration tests spent hours at `reviewDiagnostics()`. I downloaded the logs in case they expire later: 
[15_Test Project.txt](https://github.com/user-attachments/files/22430175/15_Test.Project.txt)

WilsonZiweiWang commented:
While the bitbake is scanning the project (`/integration-tests/project-folder`), multiple bitbake processes will spawn like the following:
```bash
zwang    1479052 12.5  0.4 208992 136632 ?       Sl   15:20   0:01 /usr/bin/python3 /home/zwang/projects/vscode-bitbake/resources/poky/bitbake/bin/bitbake-server decafbad 3 6 /home/zwang/projects/vscode-bitbake/integration-tests/project-folder/build/bitbake-cookerdaemon.log /home/zwang/projects/vscode-bitbake/integration-tests/project-folder/build/bitbake.lock /home/zwang/projects/vscode-bitbake/integration-tests/project-folder/build/bitbake.sock 0 0 None 0
```
In the meantime, the extension host is repeatedly trying to reconnect to the bitbake server:
<img width="1551" height="338" alt="image" src="https://github.com/user-attachments/assets/b54c211e-26f6-43c0-831e-7e30beff8721" />
The last attempt is when calling `reviewDiagnostics()` as shown in the log file I put in the above comment. Then it hangs forever as the bitbake server process no longer appears.

deribaucourt commented:
Maybe we need to disable some test parallelization option that is new or reworded?

WilsonZiweiWang committed:
chore: update extension tests path

WilsonZiweiWang committed:
chore: update extension tests path

deribaucourt committed:
integration-test: use workspace folder variable for env script path

Follow the same pattern as for other settings to avoid potential
mismatches.

deribaucourt committed:
integration-test: use workspace folder variable for env script path

Follow the same pattern as for other settings to avoid potential
mismatches.

deribaucourt committed:
integration-test: parse: allow other diagnostics to be present

With the updated extensions, some diagnostics linger. They are not
harmful, we just need the test to filter out our own expected.

deribaucourt committed:
integration-test: parse: allow other diagnostics to be present

With the updated extensions, some diagnostics linger. They are not
harmful, we just need the test to filter out our own expected.

deribaucourt committed:
integration-test: parse: allow other diagnostics to be present

With the updated extensions, some diagnostics linger. They are not
harmful, we just need the test to filter out our own expected.

deribaucourt committed:
integration-test: update for pylint extansion behavior changes

It looks like pylance now sends the diagnostics as pylint, and
require to show and save the document before seding them.

deribaucourt committed:
integration-test: update for pylint extansion behavior changes

It looks like pylance now sends the diagnostics as pylint, and
require to show and save the document before seding them.

deribaucourt committed:
integration-test: update for pylint extansion behavior changes

It looks like pylance now sends the diagnostics as pylint, and
require to show and save the document before seding them.

deribaucourt committed:
integration-test: enforce empty command wrapper setting after tests

Otherwise it would simply remove the line, which may revert to
the user's command wrapper if they had one.

deribaucourt committed:
integration-test: enforce empty command wrapper setting after tests

Otherwise it would simply remove the line, which may revert to
the user's command wrapper if they had one.

deribaucourt committed:
integration-tests: force edits to generate diagnostisc

The latest vscode version doesn't want to fire the OpenDocument
language server event, so we need to deploy full force edits to
trigger the diagnostics.
It was fine except on headless mode.

Also sometimes the diagnostics are sourced as Pylance, sometimes as
Pylint, so we use a broader filter.

See: https://github.com/microsoft/vscode/issues/138924

deribaucourt committed:
integration-tests: force edits to generate diagnostisc

The latest vscode version doesn't want to fire the OpenDocument
language server event, so we need to deploy full force edits to
trigger the diagnostics.
It was fine except on headless mode.

Also sometimes the diagnostics are sourced as Pylance, sometimes as
Pylint, so we use a broader filter.

See: https://github.com/microsoft/vscode/issues/138924

deribaucourt committed:
wip: add AI debugging logs to investigate CI issues

deribaucourt committed:
wip: add AI debugging logs to investigate CI issues

deribaucourt committed:
npm: minor version updates

Run `npm upgrade` in all packages to bump the minor revisions on all
components. Some npm audit warnings are still present so another
ncu round will be required.

deribaucourt committed:
npm: minor version updates

Run `npm upgrade` in all packages to bump the minor revisions on all
components. Some npm audit warnings are still present so another
ncu round will be required.

gcomneno committed:
test: preserve integration fixes after rebase

gcomneno committed:
test: avoid saving fixtures during document analysis

gcomneno committed:
test: warm embedded documents for diagnostics and code actions

deribaucourt committed:
github: update user id when passing DBUS address

deribaucourt committed:
Revert "wip: add AI debugging logs to investigate CI issues"

This reverts commit 58a137289605e0febaa4c4b3a55a5ec2ea941cae.

deribaucourt committed:
qemu: add script to test running in headless mode

To investigate some CI issues, I had AI create this script to run
the tests in reproducible headless QEMU VM.

deribaucourt committed:
wip: fix compile error
